### PR TITLE
Add support for get parameters

### PR DESCRIPTION
**Pull request by marianeaj**

### CODE REVIEW

File: lib/monocle.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ var _handle = function(method, path, options) {
             // Check cache if attempting to get resource
             var cached = this._cache.get(path);
             if (cached) {
-                if (!options || !options.props) {
+                if (!options || !options.props || !options.query) {
```
Comment (djvirgen) [inline lib/monocle.js]: If you pass in `options.props` but no `options.query`, this condition will be satisfied and the symlink will be resolved without props. Logic should be:

    (!options || (!options.props && !options.query))

File: lib/monocle.js [hunk 2]
```diff
@@ -129,6 +129,13 @@ var _handle = function(method, path, options) {
                         break;
                     }
                 }
+                for (var i in options.query) {
```
Comment (djvirgen) [inline lib/monocle.js]: What if `options.query` is `undefined`?

File: lib/monocle.js [hunk 3]
```diff
@@ -180,8 +187,16 @@ var buildFullPath = function(base, path) {
 };
 
 function buildQuery(options) {
-    var query = {};
+    var query = {}
+
+    if (options && options.query && typeof options.query === "object") {
+        for (var i in options.query) {
```
Comment (djvirgen) [inline lib/monocle.js]: Use [`querystring.stringify()`](https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options) to ensure keys/values are properly encoded.

File: lib/monocle.js (hunk 2 above)
Comment (marianeaj) [inline lib/monocle.js]: it won't through an error this for loop will just not be executed

Comment (djvirgen) [inline lib/monocle.js]: OK

**Comment by djvirgen** [comment]
:cherries:

**Pull request merged**